tests/integration/api/test_api.py

```diff
@@ -258,147 +157,93 @@ def test_timeboard(self):
             }
         }
 
-        timeboard_id = dog.Timeboard.update(timeboard_id, title='updated api timeboard',
-                                            description='my updated api timeboard',
-                                            graphs=[graph])['dash']['id']
-
-        # Query and ensure all is well.
-        remote_timeboard = dog.Timeboard.get(timeboard_id)
-
-        eq('updated api timeboard', remote_timeboard['dash']['title'])
-        eq('my updated api timeboard', remote_timeboard['dash']['description'])
-
-        p = graph['definition']['requests']
-        eq(p, remote_timeboard['dash']['graphs'][0]['definition']['requests'])
-
-        # Query all dashboards and make sure it's in there.
+        timeboard = dog.Timeboard.update(
+            timeboard["dash"]["id"],
+            title="updated api timeboard",
+            description="my updated api timeboard",
+            graphs=[graph]
+        )
 
-        timeboards = dog.Timeboard.get_all()['dashes']
-        ids = [timeboard["id"] for timeboard in timeboards]
-        assert timeboard_id in ids or str(timeboard_id) in ids
+        assert "updated api timeboard" == timeboard["dash"]["title"]
+        assert "my updated api timeboard" == timeboard["dash"]["description"]
+        assert timeboard["dash"]["graphs"][0] == graph
 
-        dog.Timeboard.delete(timeboard_id)
+        # Query all dashboards and make sure it"s in there.
+        timeboards = dog.Timeboard.get_all()["dashes"]
+        ids = [str(timeboard["id"]) for timeboard in timeboards]
+        assert str(timeboard["dash"]["id"]) in ids
 
-        try:
-            dog.get(timeboard_id)
-        except Exception:
-            pass
-        else:
-            # the previous get *should* throw an exception
-            assert False
+        assert dog.Timeboard.get(timeboard["dash"]["id"])["dash"]["id"] == timeboard["dash"]["id"]
+        dog.Timeboard.delete(timeboard["dash"]["id"])
+        assert "errors" in dog.Timeboard.get(timeboard["dash"]["id"])
 
     def test_search(self):
-        results = dog.Infrastructure.search(q='e')
-        assert len(results['results']['hosts']) > 0
-        assert len(results['results']['metrics']) > 0
+        results = dog.Infrastructure.search(q="e")
```

Comment:
Is this going to be flaky? Assumes we always have a host up with `e`?

Reply:
passing empty string instead. Seems to search for everything